### ISSUE TITLE
collapse_isoforms_by_sam.py giving me SyntaxError at multiple points

### ISSUE THREAD

**Issue by philoel** [Magdoll/cDNA_Cupcake#200]
Hi! First time user of this package, actually first time user of isoseq data (just got the sequencing results back from running it on some 10X GENOMICS captured cells for which we already have illumina sequencing data. Excited to compare!).

(FYI, I cloned the git repository yesterday, 22 March 2022)

I'm functional but relatively new to commandline work, and totally inexperienced with python so debugging python scripts is a bit beyond me, beyond the approach I outline below. 

I am running the command to collapse isoforms by sam just as written in the vignette, with the code 

```
collapse_isoforms_by_sam.py --input dedup.fasta \
      -s dedup.fasta.sorted.sam -c 0.99 -i 0.95 \
      --gen_mol_count
      -o dedup.5merge
```
However, I have gotten a series of SyntaxErrors (just one at a time until I push the problem under the rug, then the next error shows up). 

For the first three errors, I was able to "put the problem under the rug" by commenting out the offending line, since it was simply a command to print to stout and sterr an update on where we were. Here is an example of the first error that came up:

> Traceback (most recent call last):
>   File "/home/oel/.conda/envs/isoseq/bin/collapse_isoforms_by_sam.py", line 4, in <module>
>     __import__('pkg_resources').run_script('cupcake==28.0.0', 'collapse_isoforms_by_sam.py')
>   File "/home/oel/.conda/envs/isoseq/lib/python2.7/site-packages/pkg_resources/__init__.py", line 666, in run_script
>     self.require(requires)[0].run_script(script_name, ns)
>   File "/home/oel/.conda/envs/isoseq/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1461, in run_script
>     code = compile(source, script_filename, 'exec')
>   File "/g/arendt/Phil/condas/envs/isoseq/lib/python2.7/site-packages/cupcake-28.0.0-py2.7-linux-x86_64.egg/EGG-INFO/scripts/collapse_isoforms_by_sam.py", line 66
>     print("Picking representative sequence for {0}".format(pb_id), file=sys.stdout)
>                                                                                                                     ^ (the little hat should be under "file=sys.stdout")
> SyntaxError: invalid syntax

So, I went into the script and commented out the command to print(). 

Then, I got the same issue for lines 173, then for 183 when I "solved" each by commenting out the command to print. 

Finally, I got to line 248 which gave the following error, again a SyntaxError but this time not from a print() command so my naive solution won't work. 

> Traceback (most recent call last):
>   File "/home/oel/.conda/envs/isoseq/bin/collapse_isoforms_by_sam.py", line 4, in <module>
>     __import__('pkg_resources').run_script('cupcake==28.0.0', 'collapse_isoforms_by_sam.py')
>   File "/home/oel/.conda/envs/isoseq/lib/python2.7/site-packages/pkg_resources/__init__.py", line 666, in run_script
>     self.require(requires)[0].run_script(script_name, ns)
>   File "/home/oel/.conda/envs/isoseq/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1461, in run_script
>     code = compile(source, script_filename, 'exec')
>   File "/g/arendt/Phil/condas/envs/isoseq/lib/python2.7/site-packages/cupcake-28.0.0-py2.7-linux-x86_64.egg/EGG-INFO/scripts/collapse_isoforms_by_sam.py", line 248
>     f_group.write(f"{last_pbid}\t{members}\n")
>                                                                      ^ (the little hat is under "\n")
> SyntaxError: invalid syntax

I'm working with Python 2.7.18 in this conda environment, I believe I spotted somewhere during install that all these programs need 2.7 only. 

Any tips? I'm stuck at this point until I can get this to work, I think.

**Comment by philoel** [comment]
*cough* update: I was very, very certain that I had read that I needed python 2.7 for everything, and I did not double check that certainty. Turns out that all the parts that were giving errors were all f-strings, which apparently were implemented in 3.6. 

So, I looked at the Installing Cupcake information again and you have very prominently posted that we need python 3.7 for this. 

I am currently re-doing things and don't know if it solves my problem, but if it doesn't then surely the issue is on my end for now...

Closing in shame.
